base/src/main/java/com/smartdevicelink/managers/file/BaseFileManager.java

```diff
@@ -329,6 +329,16 @@ public void uploadFile(@NonNull final SdlFile file, final CompletionListener lis
 			listener.onComplete(true);
 			return;
 		}
+		// In sdl_ios: HAX: [#827](https://github.com/smartdevicelink/sdl_ios/issues/827) Older versions of Core had a bug where list files would cache incorrectly.
+		// This led to attempted uploads failing due to the system thinking they were already there when they were not.
+		if (!file.isPersistent() && !hasUploadedFile(file)) {
```

Comment:
In java suite, this not necessary since we are covering that case in `hasUploadedFile()`. So that if statement can be removed. This also applies to `uploadFiles()`